### PR TITLE
image_zfs: destroy correct zpool on cleanup

### PR DESCRIPTION
**Pull request by rob-wing**
As of [image_zfs: Allow same-as-host pool names](https://github.com/freebsd/poudriere/commit/73b3a5169b640460c9bb8cbd295764e9e74838ab), if an error is encountered while building an image via `image_zfs.sh`, it's possible the wrong pool will be destroyed on cleanup. This happens because the on-disk name is used when destroying the pool instead of the in-core name.

### CODE REVIEW

File: src/share/poudriere/image_zfs.sh [hunk 1]
```diff
@@ -38,10 +38,11 @@ _zfs_writereplicationstream()
 
 zfs_check()
 {
+	zroot="${ZFS_POOL_NAME}.$(jot -r 1 1000000000)"
```
Comment (jlduran) [inline src/share/poudriere/image_zfs.sh:41]: While here, what happens if someone overrides (e.g., via a pre-script) this `zfs_check()` function and "forgets" to set `zroot`?

**Review by jlduran — COMMENTED** [review]
Nice, good catch. Thank you!

**Comment by patmaddox** [comment]
I'll try running this over the next week. I had several failures today (on poudriere-devel from ports) where the message was "unable to destroy zroot" - which is a good thing it was unable, since that's my root pool! Then poudriere and any zfs commands would hang, and I had to reboot. So, I'm hopeful that this addresses that.

**Comment by patmaddox** [comment]
I'm happy to report that using this branch, poudriere cleans up the correct zpool and doesn't try to destroy my zroot! Thank you :rocket:

**Comment by allanjude** [comment]
ping

**Pull request merged**